File: eCommerce/eCommerce.Services/Database/RewardRule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCommerce.Services.Database
{
    public class RewardRule
    {
        public int RewardRuleId { get; set; }
        public int ActivityId { get; set; }
        public Activity Activity { get; set; }
        public string RewardTitle { get; set; }
        public int MaxDaysToComplete { get; set; }
        public int NumberOfPoints { get; set; }

    }
}

File: eCommerce/UI/ecommerce_mobile/lib/model/rewardrule.dart
import 'package:ecommerce_mobile/model/activity.dart';
import 'package:json_annotation/json_annotation.dart';

part 'rewardrule.g.dart';

@JsonSerializable()
class RewardRule {
  int? rewardRuleId;
  int? activityId;
  Activity? activity; // Promijeni u Activity ako imaš model
  String? rewardTitle;
  int? maxDaysToComplete;
  int? numberOfPoints;

  RewardRule({
    this.rewardRuleId,
    this.activityId,
    this.activity,
    this.rewardTitle,
    this.maxDaysToComplete,
    this.numberOfPoints,
  });

  factory RewardRule.fromJson(Map<String, dynamic> json) => _$RewardRuleFromJson(json);

  Map<String, dynamic> toJson() => _$RewardRuleToJson(this);
}
